Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPITabsCocoa.mm

```diff
@@ -428,6 +431,28 @@
     return true;
 }
 
+bool WebExtensionAPITabs::parseSendMessageOptions(NSDictionary *options, std::optional<WebExtensionFrameIdentifier>& frameIdentifier, NSString *sourceKey, NSString **outExceptionString)
+{
+    static NSDictionary<NSString *, id> *types = @{
+        frameIdKey: NSNumber.class,
+    };
+
+    if (!validateDictionary(options, sourceKey, nil, types, outExceptionString))
+        return false;
+
+    if (NSNumber *frameNumber = objectForKey<NSNumber>(options, frameIdKey)) {
+        auto identifier = toWebExtensionFrameIdentifier(frameNumber.doubleValue);
+        if (!identifier) {
+            *outExceptionString = toErrorString(nil, frameIdKey, @"it is not a frame identifier");
```

Comment:
Sometimes `outExceptionString` is +1 Retained, sometimes it is 0 Retained, which will either lead to a leak or to an over-release.